### PR TITLE
pull

### PR DESCRIPTION
**Pull request by Emileen**

### CODE REVIEW

File: resources/templates/message.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<html>
+<body>
+
+<form action="/postMessage" method="post">
+    <input type="text" placeholder="Write your message" name="Message"/>
+    <button type="submit">submit</button>
+</form>
+Message by {{name}}!
+{{name.message}}
```
Comment (raynjamin) [inline resources/templates/message.html]: Emileen, this is not the html I gave you.

File: src/com/theironyard/charlotte/Main.java [hunk 2]
```diff
@@ -1,8 +1,74 @@
...
+        // write your code here
+        Spark.init();
+
+
+        Spark.get("/", ((request, response) -> {
+                    HashMap<String, String> home = new HashMap();
+                    if (user == null) {
+                        return new ModelAndView(home, "index.html");
+                    } else {
+                        home.put("name", user.name);
```
Comment (raynjamin) [inline src/com/theironyard/charlotte/Main.java]: Good so far. Hint: more needs to be added on this step.

File: src/com/theironyard/charlotte/Main.java [hunk 3]
```diff
@@ -1,8 +1,74 @@
...
+                        return new ModelAndView(home, "message.html");
+
+                    }
+
+                }),
+                new MustacheTemplateEngine());
+
+
+        Spark.get("/postMessage", (((request, response) -> {
```
Comment (raynjamin) [inline src/com/theironyard/charlotte/Main.java]: Immediately this rings oddly- we are setting up a GET request called postMessage. Seems strange.

File: src/com/theironyard/charlotte/Main.java [hunk 4]
```diff
@@ -1,8 +1,74 @@
...
+                messageStore.put("message", user.message);
+                return new ModelAndView(messageStore, "message.html");
+
+
+            }
+
+        })), new MustacheTemplateEngine());
+
+
+        Spark.post("/login",
```
Comment (raynjamin) [inline src/com/theironyard/charlotte/Main.java]: This is perfect.

File: src/com/theironyard/charlotte/Main.java [hunk 5]
```diff
@@ -1,8 +1,74 @@
...
+                    user = new User(name);
+                    response.redirect("/");
+                    return "";
+
+                })
+
+
+        );
+
+        Spark.post("/postMessage", (((request, response) -> {
```
Comment (raynjamin) [inline src/com/theironyard/charlotte/Main.java]: here, we're posting to postmessage! I like this idea more.

File: src/com/theironyard/charlotte/Main.java [hunk 6]
```diff
@@ -1,8 +1,74 @@
...
+            String message = request.queryParams("Message");
+            user = new User(message);
+            response.redirect("/postMessage");
```
Comment (raynjamin) [inline src/com/theironyard/charlotte/Main.java]: We're posting a new message but we're making a new user here. Then we are telling the browser to redirect to `/postMessage`. What is the purpose of that new `/postMessage` page?

File: src/com/theironyard/charlotte/User.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    String name;
+    ArrayList<String> message;
+
+
+
+    public User(String name) {
+        this.name = name;
+        this.message = new ArrayList<>();
+    }
+
```
Comment (raynjamin) [inline src/com/theironyard/charlotte/User.java]: Extra white space.

**Comment by Emileen** [comment]
I did not push the homework last night

Can you please regrade this…

**Pull request closed without merging**